docs/features/kubernetes/authentication.md

```diff
@@ -0,0 +1,35 @@
+---
+id: kubernetes-auth
+title: Kubernetes Authentication
+description: Authentication in kubernetes plugin
+---
+
+The authentication process in kubernetes is basically separate from backstage auth, the
```

Comment:
As a Backstage integrator, why is it important for me to know that "kubernetes is basically separate from backstage"? Maybe we can give more details or an example? The [overall auth docs](https://backstage.io/docs/auth/) mention

> The authentication system in Backstage serves two distinct purposes: sign-in and identification of users, _as well as delegating access to third-party resources_

(emphasis mine), and I think we're talking about how 'auth' in this context is about 'access delegation' rather than 'sign-in and identification of users' -- am I understanding your intention correctly? If a Backstage integrator didn't know these things were different, maybe they would be confused because there is an `azure` auth provider for kubernetes clusters, but a `microsoft` auth provider for sign-in?

Reply:
So technically the `microsoft` provider and `azure` provider are completely different and I shouldn't have assumed configuring the first one would contribute to the later one? that is one of the points that confused me at the start, only later I got to know the `azure` one requires you to login with the `az` CLI.
How about this? 
```
The authentication process in kubernetes relies on providers, which are not the same as the application's auth providers, the default providers are defined in `plugins/kubernetes/src/kubernetes-auth-provider/KubernetesAuthProviders.ts`, you can add custom providers there if needed.

These providers are configured so your kubernetes plugin can locate and access the clusters you
have access to, some of them have special requirements in the third party in question, like Azure's Managed AAD subscription or Azure RBAC support active on the cluster.

The providers currently available are divided into server side and
client side.
```

Reply:
I like your updated suggestion @luchillo17 as it also tells the user that there's the possibility to create a custom provider if need be.

I'd probably rephrase the start slightly.

Perhaps something like this?

>The authentication process in the Kubernetes plugin relies on `KubernetesAuthProviders` for user authentication. These are not the same as the auth providers that authenticate users to Backstage itself. The default provider.....

Reply:
Done